axlearn/common/learner.py

```diff
@@ -444,6 +444,153 @@ def _mask_tree(tree: dict, *, keep: dict) -> dict:
     )
 
 
+class MetricsAccumulationOp(NamedTuple):
+    microbatches: int
+
+    def aggregrate(self, x, buffer):
+        raise NotImplementedError(self)
+
+    def normalize(self, buffer):
+        raise NotImplementedError(self)
+
+
+class ArithmeticMeanStrategy(MetricsAccumulationOp):
+    def aggregrate(self, x, buffer):
+        return buffer + x
+
+    def normalize(self, buffer):
+        return buffer / self.microbatches
+
+
+class GeometricMeanStrategy(MetricsAccumulationOp):
+    def aggregrate(self, x, buffer):
+        return buffer * x
+
+    def normalize(self, buffer):
+        return buffer ** (-self.microbatches)
+
+
+class AddStrategy(MetricsAccumulationOp):
+    def aggregrate(self, x, buffer):
+        return buffer + x
+
+    def normalize(self, buffer):
+        return buffer
+
+
+class AccumulatedLearner(Learner):
+    """Learner module that uses gradient accumulation"""
+
+    @config_class
+    class Config(Learner.Config):
+        """Configures Learner."""
+
+        # The number of microbatches to accumulate per optimizer step.
+        microbatches: Required[int] = REQUIRED
+        # tuple of key-value pairs specifying custom aggregation and normalization
+        # for a specific metric
+        metrics_accumulation_key_ops: Sequence[Dict[str, Optional[MetricsAccumulationOp]]] = []
+        gradient_dtype: Optional[jnp.dtype] = jnp.bfloat16
```

Comment:
Does the existing learner class use this? If not, we should try to be consistent with its API.

Reply:
Could you please be a little more specific, is the concern naming of variables?